cmd/notation/policy.go

```diff
@@ -0,0 +1,323 @@
+	if err != nil {
+		return err
+	}
+
+	policies := policyDocument.ListPoliciesWithinScope(scope)
+
+	return ioutil.PrintPolicyMap(os.Stdout, policies)
+}
+
+func runDeleteCommand(cmd *cobra.Command, policyNames []string) error {
+	policyDocument, err := verification.LoadDefaultPolicyDocument()
+	if err != nil {
+		return err
+	}
+
+	if err = policyDocument.DeletePolicies(policyNames, dir.UserLevel); err != nil {
+		return err
+	}
+
+	ioutil.PrintDeletedPolicyNames(os.Stdout, policyNames)
+	return nil
+}
+
+func updatePolicy(cmd *cobra.Command, opts *policyOpts) error {
+	policyDocument, err := verification.LoadDefaultPolicyDocument()
+	if err != nil {
+		return err
+	}
+	opts.prepareForUpdate()
+
+	newPolicies, err := loadPoliciesFromOptions(opts)
+	if err != nil {
+		return err
+	}
+
+	if err = policyDocument.UpdatePolicies(newPolicies, dir.UserLevel); err != nil {
+		return err
+	}
+
+	ioutil.PrintPolicyNames(os.Stdout, newPolicies, "Updated Policies:")
+	return nil
+}
+
+func addPolicy(cmd *cobra.Command, opts *policyOpts) error {
+	policyDocument, err := verification.LoadDefaultPolicyDocument()
+	if err != nil {
+		return err
+	}
+
+	policies, err := loadPoliciesFromOptions(opts)
+	if err != nil {
+		return err
+	}
+
+	if err = policyDocument.AddPolicies(policies, dir.UserLevel); err != nil {
+		return err
+	}
+
+	ioutil.PrintPolicyNames(os.Stdout, policies, "Added Policies:")
+	return nil
+}
+
+func loadPoliciesFromOptions(opts *policyOpts) ([]*verification.TrustPolicy, error) {
+	if opts.configPath == "" {
+		// create a policy from options.
+		policy, err := newPolicyFromOptions(opts)
+		if err != nil {
+			return nil, err
+		}
+		return []*verification.TrustPolicy{policy}, nil
+	}
+	// load policies from provided JSON file.
+	return verification.LoadTrustPolicies(opts.configPath)
+}
+
+func newPolicyFromOptions(opts *policyOpts) (*verification.TrustPolicy, error) {
+	override := make(map[string]string)
+	// initialize override map from the input if provided.
+	for _, config := range opts.override {
+		idx := strings.Index(config, ":")
+		if idx == -1 {
+			return nil, fmt.Errorf("invalid override config: %s", config)
+		}
+		override[config[0:idx]] = config[idx+1:]
+	}
+
+	return &verification.TrustPolicy{
+		Name:           opts.name,
+		RegistryScopes: opts.scopes,
+		SignatureVerification: verification.SignatureVerification{
+			Level:    opts.level,
+			Override: override,
+		},
+		TrustStores:       opts.stores,
+		TrustedIdentities: opts.identities,
+	}, nil
+}
+
+func (opts *policyOpts) ApplyFlags(fs *pflag.FlagSet) {
+	fs.StringSliceVar(&opts.scopes, "scope", nil, "registry scopes")
+	fs.StringVar(&opts.level, "level", "", "verification level")
+	fs.StringSliceVar(&opts.override, "level-override", nil, "config of custom verification level")
+	fs.StringSliceVar(&opts.stores, "trust-store", nil, "trust stores containing trusted roots")
+	fs.StringArrayVar(&opts.identities, "identity", nil, "trusted identities")
```

Comment:
Question for default values, currently they are just zero values, but most of them are Required fields in the spec to set up a trust policy. Looks like we could early break if user ignored one of those Required fields.

Reply:
There is a problem if we set the default values. For the update/add commands, it accepts both a file path to a config file and flags to pass in a policy. The default values only applies to the flag case not the file case. Actually, maybe we can add those default values while we create a policy from flags https://github.com/notaryproject/notation/pull/333/files#diff-25fb7823289ebe97fc1ecc1e5fb30a9799684356ddf8f2e5d0bef4b0006300c1R286. What do you think? @patrickzheng200

Reply:
> There is a problem if we set the default values. For the update/add commands, it accepts both a path to a config file and flags to pass in a policy. The default values only apply to the flag case not the file case. Actually, maybe we can add those default values while we create a policy from flags https://github.com/notaryproject/notation/pull/333/files#diff-25fb7823289ebe97fc1ecc1e5fb30a9799684356ddf8f2e5d0bef4b0006300c1R286. What do you think? @patrickzheng200

Or we just check if user provided those values in the flag case, if one of these flag is not set by the user, we could early break in the flag case.

Reply:
right, my current implementation works in this way.